@xen-orchestra/lite/src/stores/xen-api.store.ts

```diff
@@ -15,7 +18,32 @@ enum STATUS {
 }
 
 export const useXenApiStore = defineStore("xen-api", () => {
-  const xenApi = new XenApi(HOST_URL);
+  // undefined not correctly handled. See https://github.com/vueuse/vueuse/issues/3595
+  const masterSessionStorage = useSessionStorage<null | string>("master", null);
+  const router = useRouter();
+  const route = useRoute();
+
+  const masterQueryParam = computed(
+    () => route.query["master"] as string | undefined
+  );
+
+  watchEffect(() => {
+    if (masterQueryParam.value !== undefined) {
+      masterSessionStorage.value = masterQueryParam.value;
+    }
+  });
+
+  const hostUrl = computed(() => {
+    const url = new URL(HOST_URL);
+    if (masterSessionStorage.value !== null) {
+      url.hostname = masterSessionStorage.value;
+    }
+    return url.origin;
+  });
+
+  watch(hostUrl, () => window.location.reload());
+
+  const xenApi = new XenApi(hostUrl.value);
```

Comment:
I'd adopt a different approach.

1. Whenever a master is set in the URL, we store it, remove it from URL, then reload the page.
2. We build the URL (which can be static here, no need for `computed` since the page is reloaded if it changes)
3. We define `isPoolOverride` if the original URL and the current URL differ. This prevent to display a warning when the original master host is set manually in the URL

```suggestion
  whenever(
    () => route.query.master,
    async (newMaster) => {
      masterSessionStorage.value = newMaster as string;
      await router.replace({ query: { ...route.query, master: undefined } });
      window.location.reload();
    }
  );

  const hostUrl = new URL(HOST_URL);

  if (masterSessionStorage.value !== null) {
    hostUrl.hostname = masterSessionStorage.value;
  }

  const isPoolOverridden = hostUrl.origin !== new URL(HOST_URL).origin;

  const xenApi = new XenApi(hostUrl.origin);
```